File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-cd-5-4/AdjustedPaymentDates.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class AdjustedPaymentDates : ISerialized
    {
        public AdjustedPaymentDates(XmlNode xmlNode)
        {
            XmlNode adjustedPaymentDateNode = xmlNode.SelectSingleNode("adjustedPaymentDate");
            
            if (adjustedPaymentDateNode != null)
            {
                if (adjustedPaymentDateNode.Attributes["href"] != null || adjustedPaymentDateNode.Attributes["id"] != null) 
                {
                    if (adjustedPaymentDateNode.Attributes["id"] != null) 
                    {
                        adjustedPaymentDateIDRef_ = adjustedPaymentDateNode.Attributes["id"].Value;
                        XsdTypeDate ob = new XsdTypeDate(adjustedPaymentDateNode);
                        IDManager.SetID(adjustedPaymentDateIDRef_, ob);
                    }
                    else if (adjustedPaymentDateNode.Attributes["href"] != null)
                    {
                        adjustedPaymentDateIDRef_ = adjustedPaymentDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        adjustedPaymentDate_ = new XsdTypeDate(adjustedPaymentDateNode);
                    }
                }
                else
                {
                    adjustedPaymentDate_ = new XsdTypeDate(adjustedPaymentDateNode);
                }
            }
            
        
            XmlNode paymentAmountNode = xmlNode.SelectSingleNode("paymentAmount");
            
            if (paymentAmountNode != null)
            {
                if (paymentAmountNode.Attributes["href"] != null || paymentAmountNode.Attributes["id"] != null) 
                {
                    if (paymentAmountNode.Attributes["id"] != null) 
                    {
                        paymentAmountIDRef_ = paymentAmountNode.Attributes["id"].Value;
                        Money ob = new Money(paymentAmountNode);
                        IDManager.SetID(paymentAmountIDRef_, ob);
                    }
                    else if (paymentAmountNode.Attributes["href"] != null)
                    {
                        paymentAmountIDRef_ = paymentAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        paymentAmount_ = new Money(paymentAmountNode);
                    }
                }
                else
                {
                    paymentAmount_ = new Money(paymentAmountNode);
                }
            }
            
        
        }
        
    
        #region AdjustedPaymentDate_
        private XsdTypeDate adjustedPaymentDate_;
        public XsdTypeDate AdjustedPaymentDate_
        {
            get
            {
                if (this.adjustedPaymentDate_ != null)
                {
                    return this.adjustedPaymentDate_; 
                }
                else if (this.adjustedPaymentDateIDRef_ != null)
                {
                    adjustedPaymentDate_ = IDManager.getID(adjustedPaymentDateIDRef_) as XsdTypeDate;
                    return this.adjustedPaymentDate_; 
                }
                else
                {
                      return this.adjustedPaymentDate_; 
                }
            }
            set
            {
                if (this.adjustedPaymentDate_ != value)
                {
                    this.adjustedPaymentDate_ = value;
                }
            }
        }
        #endregion
        
        public string adjustedPaymentDateIDRef_ { get; set; }
        #region PaymentAmount_
        private Money paymentAmount_;
        public Money PaymentAmount_
        {
            get
            {
                if (this.paymentAmount_ != null)
                {
                    return this.paymentAmount_; 
                }
                else if (this.paymentAmountIDRef_ != null)
                {
                    paymentAmount_ = IDManager.getID(paymentAmountIDRef_) as Money;
                    return this.paymentAmount_; 
                }
                else
                {
                      return this.paymentAmount_; 
                }
            }
            set
            {
                if (this.paymentAmount_ != value)
                {
                    this.paymentAmount_ = value;
                }
            }
        }
        #endregion
        
        public string paymentAmountIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-cd-5-4/AdjustedPaymentDates.cpp
// AdjustedPaymentDates.cpp 
#include "AdjustedPaymentDates.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

AdjustedPaymentDates::AdjustedPaymentDates(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //adjustedPaymentDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* adjustedPaymentDateNode = xmlNode->FirstChildElement("adjustedPaymentDate");

   if(adjustedPaymentDateNode){adjustedPaymentDateIsNull_ = false;}
   else{adjustedPaymentDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- adjustedPaymentDateNode , address : " << adjustedPaymentDateNode << std::endl;
   #endif
   if(adjustedPaymentDateNode)
   {
      if(adjustedPaymentDateNode->Attribute("href") || adjustedPaymentDateNode->Attribute("id"))
      {
          if(adjustedPaymentDateNode->Attribute("id"))
          {
             adjustedPaymentDateIDRef_ = adjustedPaymentDateNode->Attribute("id");
             adjustedPaymentDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(adjustedPaymentDateNode));
             adjustedPaymentDate_->setID(adjustedPaymentDateIDRef_);
             IDManager::instance().setID(adjustedPaymentDateIDRef_,adjustedPaymentDate_);
          }
          else if(adjustedPaymentDateNode->Attribute("href")) { adjustedPaymentDateIDRef_ = adjustedPaymentDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { adjustedPaymentDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(adjustedPaymentDateNode));}
   }

   //paymentAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* paymentAmountNode = xmlNode->FirstChildElement("paymentAmount");

   if(paymentAmountNode){paymentAmountIsNull_ = false;}
   else{paymentAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- paymentAmountNode , address : " << paymentAmountNode << std::endl;
   #endif
   if(paymentAmountNode)
   {
      if(paymentAmountNode->Attribute("href") || paymentAmountNode->Attribute("id"))
      {
          if(paymentAmountNode->Attribute("id"))
          {
             paymentAmountIDRef_ = paymentAmountNode->Attribute("id");
             paymentAmount_ = boost::shared_ptr<Money>(new Money(paymentAmountNode));
             paymentAmount_->setID(paymentAmountIDRef_);
             IDManager::instance().setID(paymentAmountIDRef_,paymentAmount_);
          }
          else if(paymentAmountNode->Attribute("href")) { paymentAmountIDRef_ = paymentAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { paymentAmount_ = boost::shared_ptr<Money>(new Money(paymentAmountNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeDate> AdjustedPaymentDates::getAdjustedPaymentDate()
{
   if(!this->adjustedPaymentDateIsNull_){
        if(adjustedPaymentDateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDate>(IDManager::instance().getID(adjustedPaymentDateIDRef_));
        }else{
             return this->adjustedPaymentDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDate>();
   }
}
boost::shared_ptr<Money> AdjustedPaymentDates::getPaymentAmount()
{
   if(!this->paymentAmountIsNull_){
        if(paymentAmountIDRef_ != ""){
             return boost::shared_static_cast<Money>(IDManager::instance().getID(paymentAmountIDRef_));
        }else{
             return this->paymentAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Money>();
   }
}
}

